app/graphql/mutations/create_course_export.rb

```diff
@@ -1,27 +1,59 @@
 module Mutations
-  class CreateCourseExport < GraphQL::Schema::Mutation
+  class CreateCourseExport < ApplicationQuery
+    include QueryAuthorizeSchoolAdmin
+    include ValidateCourseExport
+
     argument :course_id, ID, required: true
     argument :export_type, Types::ExportType, required: true
     argument :tag_ids, [ID], required: true
     argument :reviewed_only, Boolean, required: true
+    argument :include_inactive_students, Boolean, required: true
 
-    description "Request a course export."
+    description 'Request a course export.'
 
     field :course_export, Types::CourseExportType, null: true
 
-    def resolve(params)
-      mutator = CreateCourseExportMutator.new(context, params)
+    def create_course_export
+      CourseExport.transaction do
+        tag_list = tags.present? ? tags.pluck(:name) : []
+
+        export =
+          CourseExport.create!(
+            export_type: @params[:export_type],
+            course: course,
+            user: current_user,
+            reviewed_only: !!@params[:reviewed_only],
+            include_inactive_students: @params[:include_inactive_students],
+            tag_list: tag_list
+          )
+
+        # Queue a job to prepare the report.
+        CourseExports::PrepareJob.perform_later(export)
 
-      export = if mutator.valid?
-        export = mutator.create_course_export
-        mutator.notify(:success, "Processing", "Your export is being processed. We'll notify you as soon as it is ready.")
         export
-      else
-        mutator.notify_errors
-        nil
       end
+    end
+
+    def resolve(_params)
+      notify(
+        :success,
+        'Processing',
```

Comment:
This is not yet fixed: https://github.com/pupilfirst/pupilfirst/pull/898#discussion_r803499323